src/Npgsql/PostgresErrorCodes.cs

```diff
@@ -0,0 +1,338 @@
+﻿namespace Npgsql
+{
+    /// <summary>
+    /// Provides constants to PostgreSQL's error codes.
+    /// </summary>
+    /// <remarks>
+    /// See http://www.postgresql.org/docs/current/static/errcodes-appendix.html
+    /// </remarks>
+    public static class PostgresErrorCodes
+    {
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
+
+        // Class 00 - Successful Completion
+        public const string SuccessfulCompletion = "00000";
+
+        // Class 01 - Warning
+        public const string Warning = "01000";
+        public const string DynamicResultSetsReturnedWarning = "0100C";
+        public const string ImplicitZeroBitPaddingWarning = "01008";
+        public const string NullValueEliminatedInSetFunctionWarning = "01003";
+        public const string PrivilegeNotGrantedWarning = "01007";
+        public const string PrivilegeNotRevokedWarning = "01006";
+        public const string StringDataRightTruncationWarning = "01004";
+        public const string DeprecatedFeatureWarning = "01P01";
+
+        // Class 02 - No Data
+        public const string NoData = "02000";
+        public const string NoAdditionalDynamicResultSetsReturned = "02001";
+
+        // Class 03 - SQL Statement Not Yet Complete
+        public const string SqlStatementNotYetComplete = "03000";
+
+        // Class 08 - Connection Exception
+        public const string ConnectionException = "08000";
+        public const string ConnectionDoesNotExist = "08003";
+        public const string ConnectionFailure = "08006";
+        public const string SqlclientUnableToEstablishSqlconnection = "08001";
+        public const string SqlserverRejectedEstablishmentOfSqlconnection = "08004";
```

Comment:
nit: `Sqlserver` -> `SqlServer`
nit: `Sqlclient` -> `SqlClient`
nit: `Sqlconnection` -> `SqlConnection`

Reply:
fair enough - i copied and pasted what the original author had and what the docs have it that way too

Reply:
That's what it looked like to me too. What actually piqued my interest was that the error codes seem inconsistent in this (e.g. `sqlclient` but `sql_transaction`). 

Since the goal is (at least partially) to improve readability, we might as well follow standard .NET patterns here.

Reply:
done.